gobblin-core/src/main/java/org/apache/gobblin/converter/parquet/ParquetGroup.java

```diff
@@ -0,0 +1,248 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.gobblin.converter.parquet;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+import parquet.example.data.Group;
+import parquet.example.data.simple.BinaryValue;
+import parquet.example.data.simple.BooleanValue;
+import parquet.example.data.simple.DoubleValue;
+import parquet.example.data.simple.FloatValue;
+import parquet.example.data.simple.Int96Value;
+import parquet.example.data.simple.IntegerValue;
+import parquet.example.data.simple.LongValue;
+import parquet.example.data.simple.NanoTime;
+import parquet.example.data.simple.Primitive;
+import parquet.io.api.Binary;
+import parquet.io.api.RecordConsumer;
+import parquet.schema.GroupType;
+import parquet.schema.PrimitiveType;
+
+
+/**
+ * Custom Implementation of {@link Group} to support adding any {@link PrimitiveType} and {@link GroupType}
+ * without their specific type.
```

Comment:
Just curious, what happens if a value of a different type is inserted? How many types does Parquet have?

Reply:
It should only accept PrimitiveTypes and Group and should throw an exception on any other Object type. I will fix this.

Reply:
Parquet only has primitive types. For building complex types such as map or an array, repeatable property and groups can be used on primitive types. The JSON schema is provided by the user in the source.schema property. In this way, the data types supported by the underlying data format is abstracted away from the user. The user provides schema and complex data types are built by the converter. This kind of data types abstraction can be seen in other converters as well. However, if a type defined in source.schema is not supported by the converter it will throw UnsupportedOperationException.